### PR TITLE
Generate cloud label on Prow jobs

### PR DESCRIPTION
**Pull request by hongkailiu**
Follow up https://coreos.slack.com/archives/GB7NB0CUC/p1637862410018500?thread_ts=1637860578.014800&cid=GB7NB0CUC

1. ProwGen generates the label for multistage test according to `cluster_profile`
2. This is going to be used by sanitizer to determine the cluster where the job runs.

/cc @petr-muller 

[DPTP-2651](https://issues.redhat.com/browse/DPTP-2651)

### CODE REVIEW

File: pkg/prowgen/jobbase.go
Comment (hongkailiu) [inline pkg/prowgen/jobbase.go:134]: We should consider the env var as well.

Comment (petr-muller) [reply pkg/prowgen/jobbase.go:134]: Unrelated to the envvar, could we piggyback on the `if test.MultiStageTestConfigurationLiteral.ClusterProfile != ""` condition above? Like:

```go
if clusterProfile := test.MultiStageTestConfigurationLiteral.ClusterProfile; clusterProfile != "" {
  p.PodSpec.Add(ClusterProfile(clusterProfile, test.As), LeaseClient())
  p.WithLabel(cioperatorapi.CloudLabel, clusterProfile.ClusterType())
```

I guess this is only possible if all possible clsuter profile values have a non-empty `ClusterType()`

File: pkg/prowgen/jobbase.go
Comment (petr-muller) [inline pkg/prowgen/jobbase.go:135]: One more thing, while ideating in DPTP-2649, it looks like we could also use a label that carries the actual *cluster profile* value (which is a data proxy for the "what account will the job use"), could you add it?

Comment (hongkailiu) [reply pkg/prowgen/jobbase.go:135]: Will follow up tomorrow.

**Comment by hongkailiu** [comment]
https://issues.redhat.com/browse/DPTP-2651

File: pkg/api/constant.go
Comment (petr-muller) [inline pkg/api/constant.go:23]: Let's call it `cloud-cluster-profile` so that its relation to `cluster_profile` is a little bit more obvious.

Comment (hongkailiu) [reply pkg/api/constant.go]: Updated.

**Review by petr-muller — APPROVED** [review]
/hold

Needs a followup, holding for coordinated merge ;)

**Comment by hongkailiu** [comment]
/hold cancel

I will tirgger an auto-brancher job after merge.

**Comment by hongkailiu** [comment]
Waiting for the build
https://prow.ci.openshift.org/?repo=openshift%2Fci-tools&type=postsubmit&job=branch-ci-openshift-ci-tools-master-images

**Comment by hongkailiu** [comment]
Fixing the breaking
https://github.com/openshift/release/pull/24301

**Pull request merged**
